src/components/common/ExpandableList.tsx

```diff
@@ -1,92 +1,176 @@
+      duration: 250,
       easing: Easing.inOut(Easing.circle)
     }),
     opacity: isExpanded && items.length > 0 ? sAnimationMult.value : withTiming(0, { duration: 500 })
   }))
 
-  React.useEffect(() => {
-    sAnimationMult.value = withTiming(isExpanded ? 1 : 0, {
-      duration: 500,
-      easing: Easing.inOut(Easing.circle)
-    })
-  }, [sAnimationMult, isExpanded])
+  const aFadeoutStyle = useAnimatedStyle(() => {
+    const isShowFade = items.length > maxDisplayedItems && isExpanded
+    return {
+      opacity: isShowFade ? withTiming(1, { duration: 500 }) : withTiming(0, { duration: 250 }),
+      height: isShowFade ? withTiming(48, { duration: 500 }) : withTiming(0, { duration: 250 })
+    }
+  })
+
+  const dropdownLayoutStyle = React.useMemo(
+    () =>
+      containerLayout == null
+        ? undefined
+        : {
+            left: containerLayout.x,
+            top: containerLayout.y,
+            width: containerLayout.width
+          },
+    [containerLayout]
+  )
+
+  useAsyncEffect(
+    async () => {
+      if (dropdownLayoutStyle == null || itemHeight === 0) return
+
+      sAnimationMult.value = withTiming(isExpanded ? 1 : 0, {
+        duration: 500,
+        easing: Easing.inOut(Easing.circle)
+      })
+
+      if (isExpanded && items.length > 0) {
+        await Airship.show<undefined>(bridge => {
+          setLocalBridge(bridge)
+
+          return (
+            <Animated.View style={[styles.dropdownContainer, aContainerHeightStyle, dropdownLayoutStyle]}>
+              <ShadowedView style={[styles.shadowViewStyle, Platform.OS === 'android' && styles.shadowViewStyleAndroidAdjust]}>
+                <ScrollView keyboardShouldPersistTaps="always" style={styles.scrollContainer} scrollIndicatorInsets={SCROLL_INDICATOR_INSET_FIX}>
+                  {items}
+                </ScrollView>
+                {!isExpanded ? null : (
+                  <Animated.View style={[styles.fadeoutContainer, aFadeoutStyle]}>
+                    <GradientFadeOut />
+                  </Animated.View>
+                )}
+              </ShadowedView>
+            </Animated.View>
+          )
+        })
+      } else {
+        localBridge?.resolve(undefined)
+      }
+
+      // Cleanup
+      return () => {
+        localBridge?.resolve(undefined)
+      }
+    },
+    [sAnimationMult, isExpanded, items, dropdownLayoutStyle, itemHeight],
+    'ExpandableList'
+  )
+
+  // Measure the anchor view position and width
+  React.useLayoutEffect(() => {
+    if (containerLayout == null && anchorViewRef.current != null) {
+      anchorViewRef.current.measureInWindow((x, y, width, height) => {
+        // iOS and Android return various garbage values initially.
+        // We expect positive nonzero values. We can always safely omit 0 values
+        // because we guarantee some amount of margin under correct UI design
+        if (x <= 0 || y <= 0 || width <= 0) return
```

Comment:
It seems like we don't need `useLayoutEffect` for this. Instead, we can put the measurement on line 85, right before the `Airship.show`. This means the drop-down would remain in place once it appears, even if the scene below scrolls or re-adjusts. This should be fine, especially if touching the underlying scene dismisses the drop-down.

The `useLayoutEffect` hook is kinda expensive, so we don't want to do this unless it's needed (such as right before drop-down). This might also avoid the glitches, since everything will be laid out correctly by this point.

Reply:
Ended up being a rabbit hole trying to merge it together. At least reduced this to a `useEffect` so we only fire until we get non-garbage initial values